src/include/simeng/arch/riscv/Instruction.hh

```diff
@@ -185,6 +185,9 @@ class Instruction : public simeng::Instruction {
    * processing this instruction. */
   InstructionException getException() const;
 
+  /** Retrieve exception string */
+  std::string getExceptionString() const;
+
```

Comment:
Re-looking through this, I think it can be simplified. Both the Instruction and Metadata classes contain a `exceptionString_` variable, and a `getExceptionString` function, but I don't think this is necessary. 

Given `getExceptionString` is only called in the exceptionHandler (where we get the instruction's metadata anyway) we could remove both the variable and function from the Instruction classes, and simply access `exceptionString_` via the metadata function directly.

Reply:
I agree, de-duplication would be cleaner. The proposed solution would mean we can only set the exception string from within the InstructionMetadata class as it is private e.g. we only get extra information from metadata. This would remove some potentially useful but currently unused functionality of setting the exception string from elsewhere e.g. instruction execute, pipeline units etc. This could be useful when debugging or provide more useful NYI information. There are a couple of avenues for this implementation (setter in `Instruction` to set the value in metadata or keeping the duplication) but these can be done when we want this functionality. Will de-duplicate for now unless you think differently.

Reply:
I think (and as you said) implementing extended functionality as we see fit and de-duplicating now is a sensible route. If it becomes apparent that this is very useful for debugging, or is required to be set elsewhere in the pipeline then it should be trivial to extend the functionality at that time. However, at the moment it is unknown if it will be used beyond the current case at all, so a tidier solution seems ideal